### PR TITLE
feat: 182174335 DOB was generated with unsupported format

### PR DESCRIPTION
**Pull request by zawmyolatt**

### CODE REVIEW

**Review by HJunyuan — APPROVED** [review]

File: src/models/patient.ts [hunk 1]
```diff
@@ -28,9 +35,26 @@ export const genPatientDetails = ({
...
+    ).isValid()
+  ) {
+    throw new Error(
+      `Invalid dateOfBirth (${dateOfBirth}). Should be YYYY-MM-DD or YYYY-MM or YYYY or ISO-8601 format`
+    );
+  }
+
+  const dob = dayjs(dateOfBirth, ["YYYY", "YYYY-MM", "YYYY-MM-DD"], true).isValid()
+  ? dateOfBirth
+  : dayjs.utc(dateOfBirth).format("YYYY-MM-DD");
```
Comment (zawmyolatt) [inline src/models/patient.ts:53]: no need timezone converter for this UTC date. both input date and eu offline qr dob date need to be UTC date. Shouldn't need to use dayjs timezone plugin to convert.
Note : EU required date format is like this;
Screenshot 2022-05-15 at 9 21 07 PM

File: src/models/pdt.ts [hunk 2]
```diff
@@ -82,7 +102,7 @@ export const genTestingRecord = (testingRecords: TestingRecord[], expiryDateTime
     const euTestRecord: EuTestRecord = {
       tg: targetDisease,
       tt: testingRecord.testTypeCode,
-      sc: testingRecord.collectionDateTime,
+      sc: dayjs.utc(testingRecord.collectionDateTime).format("YYYY-MM-DDTHH:mm:ss[Z]"),
```
Comment (zawmyolatt) [inline src/models/pdt.ts:105]: no need timezone converter for this UTC date. both input date and eu offline qr collectionDateTime date need to be UTC date. Shouldn't need to use dayjs timezone plugin to convert.
Note : EU required date format is like this
Screenshot 2022-05-15 at 10 11 26 PM
;

File: src/models/vac.ts [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+    const validateVaccinationDateTime = vaccination.vaccinationDateTime.split('T')[0]; // take only date value if ISO-8601 format
+    if (
+      !dayjs(
+        validateVaccinationDateTime,
+        ["YYYY-MM-DD"],
+        true
+      ).isValid()
+    ) {
+      throw new Error(
+        `Invalid vaccinationDateTime (${vaccination.vaccinationDateTime}). Should be YYYY-MM-DD format`
```
Comment (zawmyolatt) [inline src/models/vac.ts:28]: Note : EU required date format is like this;
Screenshot 2022-05-15 at 10 12 59 PM

File: src/models/rec.ts [hunk 4]
```diff
@@ -1,15 +1,64 @@
...
+    const validateFirstPositiveTestDate = recovery.firstPositiveTestDate.split('T')[0]; // take only date value if ISO-8601 format
+    if (
+      !dayjs(
+        validateFirstPositiveTestDate,
+        ["YYYY-MM-DD"],
+        true
+      ).isValid()
+    ) {
+      throw new Error(
+        `Invalid firstPositiveTestDate (${recovery.firstPositiveTestDate}). Should be YYYY-MM-DD format`
```
Comment (zawmyolatt) [inline src/models/rec.ts:27]: Note : EU required date format is like this;
Screenshot 2022-05-15 at 10 14 05 PM

File: src/models/rec.ts [hunk 5]
```diff
@@ -1,15 +1,64 @@
...
+    const validateTestValidUntil = recovery.testValidUntil.split('T')[0]; // take only date value if ISO-8601 format
+    if (
+      !dayjs(
+        validateTestValidUntil,
+        ["YYYY-MM-DD"],
+        true
+      ).isValid()
+    ) {
+      throw new Error(
+        `Invalid testValidUntil (${recovery.testValidUntil}). Should be YYYY-MM-DD format`
```
Comment (zawmyolatt) [inline src/models/rec.ts:51]: Note : EU required date format are like this;
Screenshot 2022-05-15 at 10 14 40 PM

**Review by HJunyuan — APPROVED** [review]

**Pull request merged**